src/samplers/mh.jl

```diff
@@ -0,0 +1,243 @@
+doc"""
+    HMC(n_iters::Int)
+
+Metropolis-Hasting sampler.
+
+Usage:
+
+```julia
+MH(100, (:m, (x) -> Normal(x, 0.1)))
+```
+
+Example:
+
+```julia
+# Define a simple Normal model with unknown mean and variance.
+@model gdemo(x) = begin
+  s ~ InverseGamma(2,3)
+  m ~ Normal(0,sqrt(s))
+  x[1] ~ Normal(m, sqrt(s))
+  x[2] ~ Normal(m, sqrt(s))
+  return s, m
+end
+
+sample(gdemo([1.5, 2]), MH(1000, (:m, (x) -> Normal(x, 0.1)), :s)))
+```
+"""
+immutable MH <: InferenceAlgorithm
+  n_iters   ::  Int       # number of iterations
+  proposals ::  Dict{Symbol,Any}  # Proposals for paramters
+  space     ::  Set       # sampling space, emtpy means all
+  gid       ::  Int       # group ID
+  function MH(n_iters::Int, space...)
+    new_space = Set()
+    proposals = Dict{Symbol,Any}()
+
+    # parse random variables with their hypothetical proposal
+    for element in space
+        if isa(element, Symbol)
+          push!(new_space, element)
+        else
+          @assert isa(element[1], Symbol) "[MH] ($element[1]) should be a Symbol. For proposal, use the syntax MH(N, (:m, (x) -> Normal(x, 0.1)))"
```

Comment:
Could you set up it up to do this `[MH]` automatically? Or at least make it a variable. E.g. if I subclassed MH I might want to change the text that appears in the log messages.

Reply:
good idea